code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -98,11 +110,205 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == "zombie" || name == "skeleton")
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break down some doors
+	If none of the above, play dead
+	*/
+	if(!stat)
+		if(stance = HOSTILE_STANCE_IDLE) //Not doing anything at the time
```

Comment:
Oh hell, don't tell me this is little thing was what was causing the AI to screwball